PopulationManager class #23

SlowFaller opened the pull request:
stubbed out with basic functionality #23

SlowFaller committed:
PopulationManager class

stubbed out with basic functionality

JakeGilla-DL commented on scripts/systems/population_manager.gd:
```diff
@@ -0,0 +1,76 @@
+
+@export_group("Starting values")
+@export var starting_population: int = MIN_POPULATION
+@export var starting_housed: int = MIN_POPULATION
```
This should be a separate constant.

JakeGilla-DL commented on scripts/systems/population_manager.gd:
```diff
@@ -0,0 +1,76 @@
+@export_group("Starting values")
+@export var starting_population: int = MIN_POPULATION
+@export var starting_housed: int = MIN_POPULATION
+@export var starting_hunger: int = MIN_HUNGER
```
Starting Hunger should be a seperate constant

JakeGilla-DL commented on scripts/systems/population_manager.gd:
```diff
@@ -0,0 +1,76 @@
+@export var starting_population: int = MIN_POPULATION
+@export var starting_housed: int = MIN_POPULATION
+@export var starting_hunger: int = MIN_HUNGER
+@export var starting_fulfillment: int = MIN_FULFILLMENT
```
starting fulfillment should be a seperate constants

JakeGilla-DL commented on scripts/systems/population_manager.gd:
```diff
@@ -0,0 +1,76 @@
+const MIN_FULFILLMENT = 0
+const MAX_FULFILLMENT = 100
+
+@export_group("Starting values")
```
starting values should be their own constants: for example we might want min/max hunger to be 0/100 but to start with 30 hunger.

JakeGilla-DL replied:
Once these are in a resource, these wont need to be export vars any more

JakeGilla-DL replied:
Also the init should probably assert that any starting value isnt less than the minimum or greater than the maximum.

SlowFaller replied:
Done

JakeGilla-DL commented on scripts/systems/population_manager.gd:
```diff
@@ -0,0 +1,76 @@
+extends Node
+
+const MIN_POPULATION = 3
```
We should move all these constants and the starting values below to a new resource, the read those resources in as a part of initialization

SlowFaller replied:
Done

JakeGilla-DL committed:
This is a helper class for systems that need to stack a specific material type like building resources, people, faith, food, etc.

SlowFaller committed:
Created RPopConstants resource

Abstracted starting constants for the Population class to a RPopConstants resource.

SlowFaller committed:
Merge branch '_working' of https://github.com/PlayDieRepeat/gmtk_2024 into _working

SlowFaller reviewed (commented):
Looks good,
